lib/uplink/uplink.go

```diff
@@ -16,6 +18,8 @@ import (
 
 // Config represents configuration options for an Uplink
 type Config struct {
+	log *zap.Logger
```

Comment:
this isn't exported, so no one can set it. should it be exported?

also, can you move this inside the Volatile struct?

Reply:
I'm not sure whether it needs to be exported, but I went ahead and made it exported just in case